plugins/filter/quote.py

```diff
@@ -8,15 +8,21 @@
 
 from ansible.errors import AnsibleFilterError
 from ansible.module_utils.common.collections import is_sequence
+from ansible.module_utils._text import to_text
 
-_UNSAFE_C = re.compile(r'[\s\t"]')
-_UNSAFE_CMD = re.compile(r'[\s\(\)\%\!^\"\<\>\&\|]')
+_UNSAFE_C = re.compile(u'[\\s\t"]')
+_UNSAFE_CMD = re.compile(u'[\\s\\(\\)\\^\\|%!"<>&]')
+
+# PowerShell has 5 characters it uses as a single quote, we need to double up on all of them.
+# https://github.com/PowerShell/PowerShell/blob/b7cb335f03fe2992d0cbd61699de9d9aafa1d7c1/src/System.Management.Automation/engine/parser/CharTraits.cs#L265-L272
+# https://github.com/PowerShell/PowerShell/blob/b7cb335f03fe2992d0cbd61699de9d9aafa1d7c1/src/System.Management.Automation/engine/parser/CharTraits.cs#L18-L21
+_UNSAFE_PWSH = re.compile(u"(['\u2018\u2019\u201a\u201b])")
```

Comment:
I might have missed it but doesn't this list also need to include the "standard" single quote/apostrophe `'` along with the extended 4?

Reply:
Yep, that's the first character in the regex group, after the `[`.

Reply:
ha I swear I looked it over 15 times before posting and yeah my eyes just went right over it. I figured it must have been something like that since all the tests pass.